### PR TITLE
Fix type check for non-standard schedulers in horovod

### LINKED ISSUE #14236: ReduceLROnPlateau with horovod
When I use torch.optim.lr_scheduler.ReduceLROnPlateau with horovod to train my net, horovod will check weather my lr_scheduler is pytorch_lightning.utilities.types ._LRScheduler or not, just like following (HorovodStrategy.set_up function in pytorch_lightning.strategies.horovod):
```python
        lr_scheduler_configs = self.lr_scheduler_configs
        for config in lr_scheduler_configs:
            scheduler = config.scheduler
            assert isinstance(scheduler, _LRScheduler))
            scheduler.base_lrs = [lr * self.world_size for lr in scheduler.base_lrs]
```
But，ReduceLROnPlateau does not inherent torch.optim.lr_scheduler._LRScheduler.
Anyone knows how to use ReduceLROnPlateau with horovod?

this is my optimizer and lr_scheduler in LightningModule's configure_optimizers function:
```python
        optimizer, (sch, sch_val) = get_opt_sch_bertfinetune(self, conf, self.args, None, total_steps=total_steps, val_metric_mode='min')
        # linear warm-up lr scheduler
        sch = {
            'scheduler': sch, # torch.optim.lr_schedule._LRScheduler, is ok with horovod assert check
            'interval': 'step'
        }
        # ReduceLROnPlateau
        sch_val = {
            'scheduler': sch_val, #  torch.optim.lr_scheduler.ReduceLROnPlateau
            'monitor': self.val_metric_name,
            'frequency': 1,
        }

        return [optimizer], [sch, sch_val]
```

_Originally posted by @BigodF in https://github.com/Lightning-AI/lightning/discussions/14176_

cc @awaelchli

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #14236

Fixes a typing assertion error being caused when using a special scheduler like ReduceOnPlateau with horovod. 
Discovered in a question on [stackoverflow](https://stackoverflow.com/questions/73331930/reducelronplateau-with-pytroch-lightning-and-horovod).

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @awaelchli

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]
Thank you! LGTM!
